pkg/activator/net/revision_backends.go

```diff
@@ -223,45 +223,37 @@ func (rw *revisionWatcher) probePodIPs(ready, notReady sets.String) (succeeded s
 		return rw.healthyPods, true /*no-op*/, false /* notMesh */, nil
 	}
 
-	toProbe := sets.NewString()
-
 	var healthy sets.String
 	if rw.meshMode != network.MeshCompatibilityModeAuto {
 		// If k8s marked the pod ready before we managed to probe it, and we're
 		// not also using the probe to sniff whether mesh is enabled, we can just
 		// trust k8s and mark it healthy without probing.
 		// TODO: replace with ready.Clone() once we have a recent-enough apimachinery (1.23+).
 		healthy = ready.Union(nil)
-
-		// We still want to probe non-ready pods because we can beat the kubernetes
-		// readiness propagation sometimes.
-		dests = notReady
 	} else {
-		healthy = sets.NewString()
-	}
-
-	for dest := range dests {
-		if rw.healthyPods.Has(dest) {
-			healthy.Insert(dest)
-		} else {
-			toProbe.Insert(dest)
-		}
-	}
-
-	// Short circuit case where the healthy list got effectively smaller.
-	if toProbe.Len() == 0 {
-		return healthy, false, false, nil
+		// When meshMode is auto we need to probe all pods we haven't already
+		// marked healthy - even ones in the kubernetes Ready set - in order to
+		// sniff whether mesh is enabled.
```

Comment:
This is incredibly nitty, but I got a little confused by the comment here, I think because it is phrased as which pods we need to probe, whereas the `healthy` set is actually the pods we _don't_ need to probe. 

Maybe making the comment something like 
```golang
// When meshMode is auto we need to probe all pods, so the only ones we consider 
// healthy are the ones we've already marked. We probe all other pods as we need to 
// sniff whether mesh is eanbled.
```
 would be a bit clearer.

Reply:
Had a go at clarifying this by removing the else branch entirely, so now we just add the pods Kube told us are ready IFF we're not in meshmode=auto. (And then we do the `Intersection` by removing stuff that's not in `dests` any more from the healthy set at the end, which I think is clearer anyway).